core/src/saros/versioning/VersionManager.java

```diff
@@ -45,27 +47,19 @@ public VersionCompatibilityResult determineVersionCompatibility(XMPPContact cont
     Objects.requireNonNull(contact, "contact is needed");
 
     String versionString = infoManager.getRemoteInfo(contact, VERSION_KEY).orElse(null);
+
     if (versionString == null) {
       log.warn("remote version not found for: " + contact);
       return new VersionCompatibilityResult(Compatibility.UNKNOWN, localVersion, Version.INVALID);
     }
 
     Version remoteVersion = Version.parseVersion(versionString);
+
     if (remoteVersion == Version.INVALID) {
       log.warn("contact: " + contact + ", remote version string is invalid: " + versionString);
     }
 
-    Compatibility compatibility = determineCompatibility(localVersion, remoteVersion);
+    Compatibility compatibility = localVersion.determineCompatibilityWith(remoteVersion);
```

Comment:
The result contains 2 version objects that now includes a `determineCompatibilityWith` method. I do not know if this is really a good design choice.

Reply:
I can make the method package-protected. That would keep other classes from accessing it, restricting the access to basically only `VersionManager`.